cpp/ql/src/experimental/Security/CWE/CWE-1126/DeclarationOfVariableWithUnnecessarilyWideScope.ql

```diff
@@ -0,0 +1,62 @@
+/**
+ * @name Errors When Using Variable Declaration Inside Loop
+ * @description Using variables with the same name is dangerous.
+ *              However, such a situation inside the while loop can lead to a violation of the accessibility of the program.
+ *              Requires the attention of developers.
+ * @kind problem
+ * @id cpp/errors-when-using-variable-declaration-inside-loop
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-1126
+ */
+
+import cpp
+
+/**
+ * Errors when using a variable declaration inside a loop.
+ */
+class DangerousWhileLoop extends WhileStmt {
```

Comment:
I think you could include `DoStmt` as well.

Reply:
when designing this query, I evaluated this behavior for all kinds of loops.
but `for` loops are protected by their structure from this problem.
`dowhile` loops are not subject to overriding the scope that the query is looking for.
so I have to leave only the `while`.
expansion is possible, but it will reduce the detection rate of real problems.
although I agree that overall stylistic detection will be improved.